src/s_audio.c

```diff
@@ -343,7 +339,7 @@ void sys_reopen_audio(void)
     if (as.a_api == API_PORTAUDIO)
     {
         int blksize = (as.a_blocksize ? as.a_blocksize : 64);
-        int nbufs = sys_schedadvance * as.a_srate / (blksize *1000000.);
+        int nbufs = ((sys_schedadvance / 1000.) * (as.a_srate / 1000.)) / blksize;
```

Comment:
It took me a while to figure out the difference between the two lines because of the implicit/missing casts.

The actual problem with the original code was integer overflow in `sys_schedadvance * as.a_srate` for large values of `sys_schedadvance`; e. g. with 50 ms delay: 50000 * 44100 = 2205000000 -> overflow (undefined behavior).

It could have been solved simply like this:

`(double)sys_schedadvance * as.a_srate / (blksize *1000000.)`

However, I think the following is more clear:

`int nbufs = (double)sys_schedadvance / 1000000. * as.a_srate / blksize;`

This reads as: convert `sys_schedadvance` from microseconds to seconds, multiple with the samplerate to get the number of samples and then divide by the blocksize. The explicit cast in the beginning makes it clear that we want to perform all calculations with floating point numbers (it is easy to miss the difference between `1000000` and `1000000.`)

Same thing applies to `s_audio_mmio.c`.